### DISCUSSION TITLE
Similarly Apply ValueTuple Return Syntactic Sugar To Func Return

### DISCUSSION THREAD

**Discussion by edwardjcw** [dotnet/csharplang#1527] [Language Ideas]
The new tuple return syntax (e.g., (string, int) for System.ValueTuple<string, int>) offers a readable and convenient way to use tuples. Extending that syntax to include Func<> offers an equally readable and effective approach. (string=>int) for Func<string, int>, for example.

**Comment by HaloFour** [comment]
#269 

IMO the issue with doing this is that delegates are nominally typed and not equivalent based on their signature, so `(int => bool)` would not be compatible with `Predicate<int>`.  This would make language support useless for the various APIs that don't rely on `Action<...>`/`Func<...>` either because they want a named delegate for self-documenting purposes or because they require types in the signature not supported by those delegates, such as `ref`/`out`/`in` parameters.  Tuples are quite different in that they always have the same container, `ValueTuple<...>`, which exists as a necessary evil and the compiler elides where appropriate, such as immediate construction/deconstruction.
